src/core/agent/services/ipexCommunicationService.ts

```diff
@@ -344,16 +344,43 @@ class IpexCommunicationService extends AgentService {
   }
 
   async createLinkedIpexMessageRecord(
-    credentialType: string,
-    connectionId: string,
     message: IpexMessage,
     historyType: ConnectionHistoryType
   ): Promise<void> {
+    const schemaSaid = message.exn.e.acdc.s;
+    const allSchemaSaids = Object.keys(message.exn.e.acdc?.e || {}).map(
+      // Chained schemas
+      (key) => message.exn.e.acdc.e?.[key]?.s
+    );
+    allSchemaSaids.push(schemaSaid);
+    await Promise.all(
+      allSchemaSaids.map(
+        async (schemaSaid) =>
+          await Agent.agent.connections.resolveOobi(
+            `${ConfigurationService.env.keri.credentials.testServer.urlInt}/oobi/${schemaSaid}`,
+            true
+          )
+      )
+    );
+
+    const schema = await this.props.signifyClient
+      .schemas()
+      .get(schemaSaid)
+      .catch((error) => {
+        const errorStack = (error as Error).stack as string;
+        const status = errorStack.split("-")[1];
+        if (/404/gi.test(status) && /SignifyClient/gi.test(errorStack)) {
+          return undefined;
```

Comment:
This is slightly different to what I asked.

Preferably:
- call `schemas().get()` straight away since we may already have resolved it.
- if we get a 404, try to resolve it.
  - if it resolves, we're good. If the schema is in the long running operation response, get it from there. If it's not, we can call `schemas().get()` again.
  - if it doesn't resolve, we can continue and just set undefined

Reply:
^ This will probably look a bit messy but lets see...

Reply:
> If the schema is in the long running operation response, get it from there

So we will make it wait until the long running operation is done? I think it would make our `createLinkedIpexMessageRecord` pending forever.

Reply:
Sorry I might have confused there. The "operation" returned by signify calls is a long running operation on KERIA. Any long running operations we have locally in our wallet is just tracking that long running operation in the cloud.

Here, we should call it with `waitForCompletion = true` so that it throws an error, which you can catch. This is why it will look a bit messy. If it throws the error, just catch and we don't store the credential name. (because it's not _super_ important, it's better that we just store it)